psydac/cad/tests/test_geometry.py

```diff
@@ -169,6 +171,101 @@ def test_geometry_2d_4():
     # export it
     geo.export('circle.h5')
 
+#==============================================================================
+@pytest.mark.parallel
+def test_geometry_with_mpi_dims_mask():
+
+    comm = MPI.COMM_WORLD
+    rank = comm.rank
+    size = comm.size
+    mpi_dims_mask = [False, True, False]  # We will verify that this has an effect
+    ncells = [4, 2*size, 8]  # Each process should have two cells along x2
+    degree = [2, 2, 2]
+    
+    # create an identity mapping
+    mapping = discrete_mapping('identity', ncells=ncells, degree=degree)
+
+    # create a topological domain
+    F = Mapping('F', dim=3)
+    domain = F(Cube(name='Omega'))
+
+    # associate the mapping to the topological domain
+    mappings = {domain.name: mapping}
+
+    # Define d_ncells as a dict
+    d_ncells = {domain.name: ncells}
+
+    # Create a geometry from a topological domain and the dict of mappings
+    # Here we allow for any distribution of the domain: mpi_dims_mask is not passed
+    geo = Geometry(domain=domain, ncells=d_ncells, mappings=mappings, comm=comm)
+    geo.export('geo_mpi_dims.h5')
+
+    # Read geometry file in parallel, but using mpi_dims_mask
+    geo_from_file = Geometry(filename='geo_mpi_dims.h5', comm=comm, mpi_dims_mask=mpi_dims_mask)
+
+    # Verify that the domain is distributed as expected
+    check_decomposition(geo_from_file, ncells, rank, mpi_dims_mask)
+
+    # Safely remove the file
+    comm.Barrier()
+    if rank == 0:
+        os.remove('geo_mpi_dims.h5')
+
+
+# ==============================================================================
+@pytest.mark.parallel
+def test_from_discrete_mapping():
+
+    comm = MPI.COMM_WORLD
+    rank = comm.rank
+    size = comm.size
+    mpi_dims_mask = [False, False, True]  # We swill verify that this has an effect
+    ncells = [4, 8, 2 * size]  # Each process should have two cells along x3
+    degree = [3, 3, 3]
```

Comment:
@campospinto We could compute the expected values here:
```suggestion
    degree = [3, 3, 3]

    expected_starts = [0, 0, 2 * rank]
    expected_ends   = [3, 7, 2 * rank + 1]
```